internal/sync/syncer.go

```diff
@@ -234,18 +243,25 @@ func (r *resourceInfoProviderStub) IsNamespaced(_ schema.GroupKind) (bool, error
 	return false, nil
 }
 
-func (s *Syncer) sync(gitSync *v1alpha1.GitSync, targetObjs []*unstructured.Unstructured, logger *zap.SugaredLogger) bool {
+// sync compares the live state of the watched resources to target state defined in git
+// for the given GitSync. If it doesn't match, syncing the state to the live objects. Otherwise,
+// skip the syncing.
+func (s *Syncer) sync(
+	gitSync *v1alpha1.GitSync,
+	targetObjs []*unstructured.Unstructured,
+	logger *zap.SugaredLogger,
+) (gitopsSyncCommon.OperationPhase, string) {
 	logEntry := log.WithFields(log.Fields{"gitsync": gitSync})
 
 	reconciliationResult, modified, err := s.compareState(gitSync, targetObjs)
 	if err != nil {
-		return false
+		return gitopsSyncCommon.OperationError, "error on comparing git sync state"
```

Comment:
should we log the actual `err` if we are returning the custom error message?

Reply:
yeah, I think it would be good to have the original error